src/index.ts

```diff
@@ -1,84 +1,91 @@
 import { execSync, exec } from 'child_process';
-import { getInput, info, setFailed } from "@actions/core";
+import { error, getInput, getMultilineInput, info, setFailed } from "@actions/core";
 
-const ENV = {
+const inputs = {
   email: getInput("email"),
   apiKey: getInput("api_key"),
-  devAppName: getInput("dev_app_name"),
-  // uatAppName: core.getInput("uat_app_name"),
+  appNames: getMultilineInput("app_names"),
 } as const;
 
-type Env = typeof ENV;
+const checkInputs = () => {
+  const allPresent =
+    Object.values(inputs).every((value) => !!value && !!value.length)
 
-const createNetrcFile = ({ email, apiKey }: Env) => execSync(`cat >~/.netrc <<EOF
+  if (!allPresent) {
+    throw new Error("Missing an input variable")
+  }
+}
+
+const createNetrcFile = () => execSync(`cat >~/.netrc <<EOF
 machine api.heroku.com
-    login ${email}
-    password ${apiKey}
+    login ${inputs.email}
+    password ${inputs.apiKey}
 machine git.heroku.com
-    login ${email}
-    password ${apiKey}
+    login ${inputs.email}
+    password ${inputs.apiKey}
 EOF`);
 
-const addRemotes = ({ devAppName }: Env) => {
-  const addRemote = (app: string) => {
+const addRemotes = () => {
+  const addRemote = (app: string, index: number) => {
     try {
-      info("Setting remote with Heroku CLI...")
+      info(`[App ${index}] Setting remote with Heroku CLI...`);
```

Comment:
The index isn't useful here, it's better to know which app name is being deployed.
```suggestion
      info(`[App: ${app}] Setting remote with Heroku CLI...`);
```